src/backend/app/auth/roles.py

```diff
@@ -242,6 +242,12 @@ async def wrap_check_access(
     role: ProjectRole,
 ) -> ProjectUserDict:
     """Wrap check_access call with HTTPException."""
+    # If project is public, skip permission check
+    if project.visibility == ProjectVisibility.PUBLIC:
```

Comment:
Adding this logic here means that all users have access to all endpoints if the project is public.

So they could delete the project, modify anything, etc.

It's currently only present in the `mapper` permission, as we only want to bypass the auth if the endpoint requires `mapper` role only:

https://github.com/hotosm/fmtm/blob/fd11d08f9c5c5c88c6d4b1c7394e517649f0c628/src/backend/app/auth/roles.py#L327

Was this logic needed for the code in this PR to work?

Reply:
Everything else is all set to merge!

Reply:
I did a minor update, fixing the return type on the `mapper` role if `ProjectVisibility.PUBLIC`.

I'm going to merge as this PR has been open a while - let me know if any issues!

Reply:
Uh I totally missed that. Yes , you are right. I changed that in order to get ProjectUserDict.

Reply:
Uh I totally missed that. Yes , you are right. I changed that in order to get ProjectUserDict.